components/button/button.js

```diff
@@ -0,0 +1,109 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ButtonMixin } from './button-mixin.js';
+import { buttonStyles } from './button-styles.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { VisibleOnAncestorMixin } from '../../mixins/visible-on-ancestor-mixin.js';
+
+class Button extends ButtonMixin(VisibleOnAncestorMixin(RtlMixin(LitElement))) {
+
+	static get styles() {
+		return [ buttonStyles,
+			css`
+
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			button {
+				font-family: inherit;
+				padding: 0.5rem 1.5rem;
+				@apply --d2l-button-shared;
+				@apply --d2l-label-text;
+				@apply --d2l-button;
+				@apply --d2l-button-clear-focus;
+				width: 100%;
+			}
+			/* Firefox includes a hidden border which messes up button dimensions */
+			button::-moz-focus-inner {
+				border: 0;
+			}
+			button,
+			button[disabled]:hover,
+			button[disabled]:focus,
+			:host([active]) button[disabled] {
+				background-color: var(--d2l-color-regolith);
+				border-color: var(--d2l-color-mica);
+				color: var(--d2l-color-ferrite);
+			}
+			button:hover,
+			button:focus,
+			:host([active]) button,
+			:host(.d2l-button-hover) button,

```

Comment:
I don't think we should need the `.d2l-button-hover` and `.d2l-button-focus` classes.  My understanding is that these were used for testing purposes only.  I think I was successfully able to omit them when I did the other button conversions.

Reply:
Agreed -- they were for the old Galen visual tests. 🔥